scripts/wine-pyinstaller.sh

```diff
@@ -0,0 +1,172 @@
+        "alert"
+      exit -1
+   fi
+
+   if [ -z "$(wine --version 2> /dev/null \
+              | grep "wine-${MIN_WINE_VER}")" ]; then
+      colorEcho \
+        "[-] wine minimum version ${MIN_WINE_VER} required\n" \
+        "alert"
+      exit -1
+   fi
+}
+
+stripWhitespace() {
+  local input="$@"
+  printf "${input}\n" | tr -d "[:space:]"
+}
+
+colorEcho() {
+  local str="$1"
+  local color="${COLOR_GOOD}${COLOR_BOLD}"
+
+  if [ ! -z "$2" ] \
+    && [ "$(stripWhitespace "$2")" = "alert" ]; then
+    color="${COLOR_BAD}${COLOR_BOLD}"
+  fi
+
+  printf "${color}${str}${COLOR_RESET}"
+}
+
+checkPrerequisite() {
+  local bin_name="$1"
+  local bin_loc=$(which "${bin_name}" 2>/dev/null)
+
+  if [ -z "${bin_loc}" ] \
+    || [ ! -f "${bin_loc}" ]; then
+    printf "    - '$1' not found, quitting ...\n"
+    exit -1
+  else
+    printf "    - '${bin_name}' found at ${bin_loc}\n"
+  fi
+}
+
+
+mainFunction() {
+  export "WINEARCH=${WINEARCH}"
+  export "WINEPREFIX=${WINEDIR}"
+
+
+  ## Check prerequisites
+  colorEcho "[+] Checking prerequisites ...\n"
+  for bin in ${PREREQUISITES}; do
+    checkPrerequisite "${bin}"
+  done
+
+  checkRequiredVersions
+
+
+  ## Create required directories
+  mkdir -pv "${TESTDIR}"
+  mkdir -pv "${OUTDIR}"
+  mkdir -pv "${DLDIR}"
+  mkdir -pv "${OUTDIR}"
+
+
+  colorEcho "[+] Downloading necessary files ...\n"
+  ## Download Python and Pip
+  cd "${DLDIR}"
+
+  if [ ! -f "${WINPY_EXE}" ]; then
+    wget \
+      "https://github.com/winpython/winpython/releases/download/1.10.20180404/WinPython32-${PYVER}.2Zero.exe" \
+      -O "${WINPY_EXE}"
+  fi
+
+  if [ ! "$(sha256sum "${WINPY_EXE}" \
+    | cut -d" " -f1)" = ${WINPYHASH} ]; then
+    colorEcho "[-] ${WINPY_EXE} has incorrect hash, quitting ...\n"
+  fi
+
+  wget "${GEN_NATIVE_URL}" -O "${DLDIR}/gen_native_message.py"
+  wget "${NATIVE_MAIN_URL}" -O "${DLDIR}/native_main.py"
+
+  wget "https://bootstrap.pypa.io/get-pip.py" -O "${DLDIR}/get-pip.py"
+
+  ## Extract Python-3.5.4 from WinPython
+  colorEcho "[+] Extract Python-${PYVER}\n"
+  cd "${TESTDIR}"
+  7z x "${DLDIR}/winpython-${PYVER}.exe" "python-${PYVER}" -aoa
+
+
+  ## Install Pip and PyInstaller
+  rm -rf "${WINEDIR}"
+  cd "${PYDIR}"
+
+  colorEcho "[+] Installing Pip ...\n"
+  wine python.exe "${DLDIR}/get-pip.py"
+
+  colorEcho "[+] Installing PyInstaller ...\n"
+  wine python.exe -m pip install --upgrade pyinstaller
+
+  ## Compile with PyInstaller
+  colorEcho "[+] Compiling with PyInstaller under Wine ...\n"
+  rm -rf "${OUTDIR}"
+  PYTHONHASHSEED=1 wine Scripts/pyinstaller \
+    --clean \
+    --console \
+    --onefile \
+    --noupx \
+    --noconfir \
+    --log-level=ERROR \
+    --workpath "${OUTDIR}" \
+    --distpath "${OUTDIR}" \
+    "${DLDIR}/native_main.py"
+
+  ## Test the compiled EXE
+  colorEcho "[+] Checking compiled binary ...\n"
+  OUTFILE="${OUTDIR}/native_main.exe"
+
+  if [ -f "${OUTFILE}" ]; then
+    python3 \
+      "${DLDIR}/gen_native_message.py" version \
```

Comment:
There's a few problems with this:

- gen_native_message.py needs `chmod +x` 
- gen_native_message.py has Windows line endings which breaks env:
```fish
wine-pyinstaller/downloads/gen_native_message.py native
/usr/bin/env: ‘python3\r’: No such file or directory
```
- Once I fixed all that, gen_native_message.py gives
```fish
wine-pyinstaller/downloads/gen_native_message.py native
Traceback (most recent call last):
  File "wine-pyinstaller/downloads/gen_native_message.py", line 39, in <module>
    val = sys.argv[i].strip().split(separator)[1]
IndexError: list index out of range
```
I think because you've forgotten your own syntax: If I change this to `"${DLDIR}/gen_native_message.py" cmd..version` it works:

```fish
wine-pyinstaller/downloads/gen_native_message.py cmd..version | wine wine-pyinstaller/dist/nativ
e_main.exe
{"version": "0.1.6"}0030:fixme:ver:GetCurrentPackageId (0x32fe24 (nil)): stub
```